### PR TITLE
Harshad number documentation

### LINKED ISSUE #308: Documentation for harshad number
## 📚 Documentation

Add Documentation for harshad number 

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Ayushjain2205**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation from Harshad Number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #308

### CODE REVIEW

**Comment by vybhav72954** [comment]
Link the issue properly

File: docs/mathematics/harshad-number.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="py"
+  values={[
+    { label: 'Python', value: 'py', },
+    { label: 'Dart', value: 'dart', },
```
Comment (HarshCasper) [inline docs/mathematics/harshad-number.mdx:39]: Dart is not yet supported. Please use some other language. Prefer C++ or Java

File: docs/mathematics/harshad-number.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+## Introduction
+
+A number is called a Harshad number if it is divisible by the sum of its digits. It can also be referred to as Niven number.
+
+## Explanation
+
+If the number under consideration is taken and the sum of its individual digits is calculated, if this resultant sum is divisible by the number itself then the number will be considered a Harshad number.
+
+Let's have one example:
```
Comment (HarshCasper) [inline docs/mathematics/harshad-number.mdx:14]: Give a line break

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/harshad-number.mdx (hunk 1 above)
Comment (Ayushjain2205) [reply docs/mathematics/harshad-number.mdx:39]: @HarshCasper harshad number is only there in python and dart...so is it fine to add support only for python or should I add code for it in java or c++ first and then make updates here

**Comment by vybhav72954** [comment]
@Ayushjain2205 Why have you closed the PR?

**Comment by Ayushjain2205** [comment]
@vybhav72954  while making the changes I accidently deleted my local branch so the PR got closed so I had to make  a [new PR](https://github.com/TesseractCoding/NeoAlgo-Docs/pull/483) 
Sorry for the error

**Pull request closed without merging**